log.go

```diff
@@ -80,6 +93,68 @@ type eventLogger struct {
 	// TODO add log-level
 }
 
+func (el *eventLogger) Start(ctx context.Context, name string) context.Context {
+	_, ctx = opentrace.StartSpanFromContext(ctx, name)
+	return ctx
+}
+
+func (el *eventLogger) StartFromParentState(ctx context.Context, name string, parent []byte) context.Context {
+	sc := deserializeContext(parent)
+	_, ctx = opentrace.StartSpanFromContext(ctx, name, otExt.RPCServerOption(sc))
+	return ctx
+}
+
+func (el *eventLogger) SerializeContext(ctx context.Context) []byte {
+	gTracer := opentrace.GlobalTracer()
+	b := make([]byte, 0)
+	carrier := bytes.NewBuffer(b)
+	span := opentrace.SpanFromContext(ctx)
+	if err := gTracer.Inject(span.Context(), opentrace.Binary, carrier); err != nil {
+		// TODO: handle this case without having to deal with errors when writing logs
+		return nil
+	}
+	return carrier.Bytes()
+}
+
+func (el *eventLogger) LogKV(ctx context.Context, key string, value interface{}) {
+	span := opentrace.SpanFromContext(ctx)
```

Comment:
We should avoid crashing if the context doesn't contain a span. In that case, we should probably create some form of "EventMissing" event and log to that.